pennylane/ops/op_math/decompositions/solovay_kitaev.py

```diff
@@ -125,18 +127,23 @@ def _approximate_set(basis_gates, max_length=10):
     gtrie_ids = [[[gate] for gate in basis]]
     gtrie_mat = [list(basis_mat.values())]
     gtrie_gph = [list(basis_gph.values())]
+    gtrie_sum = [[int(gate in t_set) for gate in basis]]
 
     # Maintains the approximate set for gates' names, SU(2)s, global phases and quaternions
     approx_set_ids = list(gtrie_ids[0])
     approx_set_mat = list(gtrie_mat[0])
     approx_set_gph = list(gtrie_gph[0])
+    approx_set_sum = list(gtrie_sum[0])
     approx_set_qat = [_quaternion_transform(mat) for mat in approx_set_mat]
 
     # We will perform a breadth-first search (BFS) style set building for the set
     for depth in range(max_length - 1):
+        kdtree = KDTree(qml.math.array(approx_set_qat))
         # Add the containers for next depth while we explore the current
-        gtrie_id, gtrie_mt, gtrie_gp = [], [], []
-        for node, su2m, gphase in zip(gtrie_ids[depth], gtrie_mat[depth], gtrie_gph[depth]):
+        gtrie_id, gtrie_mt, gtrie_gp, gtrie_sm, gtrie_qt = [], [], [], [], []
+        for node, su2m, gphase, tgsum in zip(
+            gtrie_ids[depth], gtrie_mat[depth], gtrie_gph[depth], gtrie_sum[depth]
+        ):
```

Comment:
This is: For each sequence's node, su2m, repr, gphase, and tgsum at depth D, do ...  

....right?

Reply:
Yes. This iterates over the data for the gate sequence in the current `depth` and then prepares the data to be used in the next `depth`.